src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -32,56 +32,82 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceLoadedDotNetRuntimeExtensions
     {
+        internal static readonly string LoadedDotNetRuntimesKey = "Computers/LoadedDotNetRuntimes";
+
         public static void NeedLoadedDotNetRuntimes(this TraceEventDispatcher source)
         {
+            if (source.userData.ContainsKey(LoadedDotNetRuntimesKey))
+            {
+                // The callbacks are already registered for this source
+                return;
+            }
+
             // ensure there are base processes
```

Comment:
> The existing semantics must be persisted, they exist for a reason.

Note that I'm not trying to claim otherwise. However, since these reasons weren't completely clear to me at the start of the work, I am interested in using the additional information from you to 1) help future readers understand these scenarios and 2) formulate some amount of automated testing to ensure the scenarios do not regress in the future.